src/Uno.UI/UI/Xaml/ResourceResolver.cs

```diff
@@ -0,0 +1,218 @@
+
+		private static readonly Dictionary<string, Func<ResourceDictionary>> _registeredDictionaries = new Dictionary<string, Func<ResourceDictionary>>();
+
+		public static class TraceProvider
+		{
+			public readonly static Guid Id = Guid.Parse("{15E13473-560E-4601-86FF-C9E1EDB73701}");
+
+			public const int InitGenericXamlStart = 1;
+			public const int InitGenericXamlStop = 2;
+		}
+
+		private readonly static IEventProvider _trace = Tracing.Get(TraceProvider.Id);
+
+		private static readonly ILogger _log = typeof(ResourceResolver).Log();
+
+		private static readonly Stack<XamlScope> _scopeStack;
+
+		/// <summary>
+		/// The current xaml scope for resource resolution.
+		/// </summary>
+		internal static XamlScope CurrentScope => _scopeStack.Peek();
+
+		static ResourceResolver()
+		{
+			_scopeStack = new Stack<XamlScope>();
+			_scopeStack.Push(XamlScope.Create()); //There should always be a base-level scope (this will be used when no template is being resolved)
+		}
+
+		/// <summary>
+		/// Performs a one-time, typed resolution of a named resource, using Application.Resources.
+		/// </summary>
+		/// <typeparam name="T"></typeparam>
+		/// <param name="key"></param>
+		/// <param name="context">Currently unused. In place to have the possibility to modify the lookup mechanism without introducing a binary breaking change.</param>
+		/// <returns></returns>
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public static T ResolveResourceStatic<T>(object key, object context = null)
+		{
+			if (TryStaticRetrieval(key, out var value) && value is T tValue)
+			{
+				return tValue;
+			}
+
+			return default(T);
+		}
+
+		internal static T ResolveTopLevelResource<T>(object key)
+		{
+			if (TryTopLevelRetrieval(key, out var value) && value is T tValue)
+			{
+				return tValue;
+			}
+
+			return default(T);
+		}
+
+		/// <summary>
+		/// Apply a StaticResource or ThemeResource assignment to a DependencyProperty of a DependencyObject. The assignment will be provisionally
+		/// made immediately using Application.Resources if possible, and retried at load-time using the visual-tree scope.
+		/// </summary>
+		/// <param name="owner">Owner of the property</param>
+		/// <param name="property">The property to assign</param>
+		/// <param name="resourceKey">Key to the resource</param>
+		/// <param name="context">Currently unused. In place to have the possibility to modify the lookup mechanism without introducing a binary breaking change.</param>
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public static void ApplyResource(DependencyObject owner, DependencyProperty property, object resourceKey, bool isThemeResourceExtension, object context = null)
+		{
+			// Set initial value based on statically-available top-level resources.
+			if (TryStaticRetrieval(resourceKey, out var value))
+			{
+				owner.SetValue(property, BindingPropertyHelper.Convert(() => property.Type, value));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'owner' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305744)